README.md

```diff
@@ -1,9 +1,72 @@
 OkHttp
 ======
 
-An HTTP & HTTP/2 client for Android and Java applications.
+See the [project website][okhttp] for documentation and APIs.
 
-See the [**project website**][website] for documentation and APIs.
+HTTP is the way modern applications network. It’s how we exchange data & media. Doing HTTP
+efficiently makes your stuff load faster and saves bandwidth.
+
+OkHttp is an HTTP client that’s efficient by default:
+
+ * HTTP/2 support allows all requests to the same host to share a socket.
+ * Connection pooling reduces request latency (if HTTP/2 isn’t available).
+ * Transparent GZIP shrinks download sizes.
+ * Response caching avoids the network completely for repeat requests.
+
+OkHttp perseveres when the network is troublesome: it will silently recover from common connection
+problems. If your service has multiple IP addresses OkHttp will attempt alternate addresses if the
+first connect fails. This is necessary for IPv4+IPv6 and for services hosted in redundant data
+centers. OkHttp supports modern TLS features (TLS 1.3, ALPN, certificate pinning). It can be
+configured to fall back for broad connectivity.
+
+Using OkHttp is easy. Its request/response API is designed with fluent builders and immutability. It
+supports both synchronous blocking calls and async calls with callbacks.
+
+
+Get a URL
+---------
+
+This program downloads a URL and prints its contents as a string. [Full source][get_example].
+
+```java
+OkHttpClient client = new OkHttpClient();
+
+String run(String url) throws IOException {
+  Request request = new Request.Builder()
+      .url(url)
+      .build();
+
+  try (Response response = client.newCall(request).execute()) {
+    return response.body().string();
+  }
+}
+```
+
+
+Post to a Server
+----------------
+
+This program posts data to a service. [Full source][post_example].
+
+```java
+public static final MediaType JSON
```

Comment:
We should fix this in 4.1, provide well known MediaTypes

Reply:
I am not a fan. Guava’s list seems like a massive hassle.

Reply:
And they begrudgingly have to keep adding more and more endlessly...

Reply:
https://github.com/google/guava/search?q=MediaType&type=Commits

Reply:
I just want JSON :)

Reply:
Yeah, but I don’t trust you to stop there, and I can’t provide a pithy rule for what we include and what we exclude.

Reply:
Rule of 3, so we should add application/vnd.openxmlformats-officedocument.spreadsheetml.sheet now